virsh_migration: Migration testcases development

balamuruhans opened the pull request:
As virsh_migrate.py grew up and difficult to maintain, added testcases
in new file for further development.

Suggested-by: Dan Zheng <dzheng@redhat.com>
Signed-off-by: Balamuruhan S <bala24@linux.vnet.ibm.com>

balamuruhans commented:
Work in progress, it depends on avocado-vt PRs,
https://github.com/avocado-framework/avocado-vt/pull/1282
https://github.com/avocado-framework/avocado-vt/pull/1283
https://github.com/avocado-framework/avocado-vt/pull/1284

chunfuwen commented:
Those PRs have been reviewed,please see related comments.

Chunfu Wen

----- Original Message -----
From: "Balamuruhan S" <notifications@github.com>
To: "autotest/tp-libvirt" <tp-libvirt@noreply.github.com>
Cc: "Subscribed" <subscribed@noreply.github.com>
Sent: Tuesday, January 9, 2018 10:54:16 PM
Subject: Re: [autotest/tp-libvirt] virsh_migration: Migration testcases development (#1362)

Work in progress, it depends on avocado-vt PRs,
https://github.com/avocado-framework/avocado-vt/pull/1282
https://github.com/avocado-framework/avocado-vt/pull/1283
https://github.com/avocado-framework/avocado-vt/pull/1284

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/autotest/tp-libvirt/pull/1362#issuecomment-356306475

balamuruhans commented:
@dzhengfy @chunfuwen I have implemented migration testcases as virsh_migrate.py and migrate_vm.py are difficult to maintain,

I have implemented and tested with single VM and with multi VMs migration, please help to review.

**Test results:**
```
JOB ID     : fab0a62ae8cb55ec1af8a7973ccb1512f9bc72c1
JOB LOG    : /home/bala/avocado-fvt-wrapper/results/job-2018-01-14T20.42-fab0a62/job.log
 (1/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_precopy.live:  PASS (309.26 s)
 (2/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_precopy.online:  FAIL (320.86 s)
 (3/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_precopy.offline:  PASS (324.64 s)
 (4/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_postcopy.postcopy:  PASS (381.58 s)
 (5/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_postcopy.postcopy_after_precopy:  PASS (387.96 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1728.97 s
JOB HTML   : /home/bala/avocado-fvt-wrapper/results/job-2018-01-14T20.42-fab0a62/results.html

```

sathnaga commented on libvirt/tests/src/virsh_cmd/domain/virsh_migration.py:
```diff
@@ -0,0 +1,172 @@
+                             "expected")
+                return
+        func("%s did not respond after %d sec." % (vm.name, ping_timeout))
+
```
@balamuruhans this method can be placed outside of this testcase file so that other migrate tests can utilise this, 
how about tune it little generic and keep it in utils_misc.py...

balamuruhans replied:
@sathnaga sure, I will work on it.

balamuruhans commented:
Also depends on https://github.com/avocado-framework/avocado-vt/pull/1304

balamuruhans commented:
Tested with fixes:
```
JOB ID     : 66cd7ed69f75eefc0dab3879d16232587522e0c4
JOB LOG    : /home/bala/avocado-fvt-wrapper/results/job-2018-01-30T17.26-66cd7ed/job.log
 (1/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_precopy.live:  PASS (511.68 s)
 (2/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_precopy.online:  PASS (211.49 s)
 (3/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_precopy.offline:  PASS (269.64 s)
 (4/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_postcopy.postcopy:  PASS (247.01 s)
 (5/5) guest_migration.qemu.qcow2.virtio_scsi.smp2.virtio_net.Guest.Pegas.1.1.ppc64le.powerkvm-libvirt.virsh.migration.migrate_postcopy.postcopy_after_precopy:  PASS (246.18 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1490.92 s
JOB HTML   : /home/bala/avocado-fvt-wrapper/results/job-2018-01-30T17.26-66cd7ed/results.html

```

balamuruhans commented:
Thanks for reviewing @sathnaga,

This is base for migration test case, the scenarios are yet to be added and I will ensure they are not covered in old files.
